docs/topic-guides/workflows/workflow-file.rst

```diff
@@ -0,0 +1,362 @@
+workflow.
+
+.. code-block:: yaml
+
+    name: flights-processing-pipeline
+    steps:
+      filter:
+        command: python $filter-py $raw-flights $filtered-flights
+        inputs:
+          - filter-py:
+              path: src/filter_flights.py
+          - raw-flights:
+              path: data/flight-data/2019-01-flights.csv.zip
+        outputs:
+          - filtered-flights:
+              path: data/output/flights-filtered.csv
+
+      count:
+        command: python $count-py $filtered-flights $flight_count
+        inputs:
+          - count-py:
+              path: src/count_flights.py
+          - filtered-flights:
+              path: data/output/flights-filtered.csv
+        outputs:
+          - flight_count:
+              path: data/output/flights-count.csv
+
+
+Executing a Workflow File
+-------------------------
+
+Running :meth:`renku run workflow.yml <renku.ui.cli.run>` will execute all steps
+in the workflow file. Executing the workflow will commit all workflow inputs and
+outputs, too, including the workflow file itself.
+
+.. code-block:: console
+
+    $ renku run workflow.yml
+    Executing step 'workflows/workflow-topic-guide-last.yml::filter': 'python src/filter_flights.py data/flight-data/2019-01-flights.csv.zip data/output/flights-filtered.csv' ...
+    Executing step 'workflows/workflow-topic-guide-last.yml::count': 'python src/count_flights.py data/output/flights-filtered.csv data/output/flights-count.csv' ...
+
+.. note:: **Do you have output files you don't want to be committed, such as log files?**
+    You have 2 options: (1) Do not list these outputs in the workflow definition
+    file, and Renku will ignore them. Or, (2) include the file in the workflow
+    file, but use the ``persist: false`` flag to tell Renku not to commit the
+    file.
+
+Executing a Portion of a Workflow
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Renku also helps you run only portions of your workflow at a time. For example,
+you can execute just one step of the workflow by referencing that step's name:
+
+.. code-block:: console
+
+        $ renku run workflow.yml filter
+
+You may specify more than one step to run:
+
+.. code-block:: console
+
+        $ renku run workflow.yml filter count
+
+.. If we had a longer workflow, perhaps with 10 or more steps, we could specify a
+.. subset of steps to run.
+
+.. .. code-block:: console
+
+..         # runs the step 'filter' and every step after it.
+..         $ renku run workflow.yml filter:
+
+..         # runs every step before 'count', and the 'count' step
+..         $ renku run workflow.yml :count
+
+..         # runs every step between 'filter' and 'count', including 'filter' and 'count' themselves
+..         $ renku run workflow.yml filter:count
+
+
+Workflow Step Execution Order
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When you execute a workflow file, Renku builds an execution graph. This means
+that Renku determines how the steps in the workflow are related. For example,
```

Comment:
I would change this to something similar to:

`When you execute a workflow file, Renku builds an execution graph to determines how the steps in the workflow are related. Renku then executes steps in that order. This means that the order of steps in the workflow file has no effect on the execution order.`

I believe we should clearly say the last sentence and also say that the execution graph is used to determine the execution order.